cantools/database/dataelement.py

```diff
@@ -0,0 +1,192 @@
+                logger.warning(
+                    f"Piecewise linear type: {self.name} has overlapping segments! "
+                    f"Segment {i} starts at phys val {scaled_segment[0]} "
+                    f"but one of the prev segments ended at {last_phys_max}. "
+                    "Encoding might be ambiguous."
+                )
+
+            last_phys_max = max(scaled_segment[1], last_phys_max)
+
+    def choice_string_to_number(self, string: str) -> int:
+        if self.choices is None:
+            raise ValueError(f"Signal {self.name} has no choices.")
+
+        for choice_number, choice_value in self.choices.items():
+            if str(choice_value) == str(string):
+                return choice_number
+
+        raise KeyError(f"Choice {string} not found in data element {self.name}.")
+
+    def _get_offset_scaling_from_list(self, val: float, segments: PiecewiseSegment) -> Tuple[float, float]:
+        for i, segment in enumerate(segments):
+            start, end = segment
+            if start <= val <= end:
+                return self._offset[i], self._scale[i]  # type: ignore
+        else:
+            err_text = [f"{start} <= x <= {end}" for start, end in segments]
+            raise ValueError(
+                f"Value {val} is not in ranges: \n {' OR '.join(err_text)}"
+            )
+
+    def get_offset_scaling_from_raw(
+        self, raw_val: Optional[float] = None
+    ) -> Tuple[float, float]:
+        """Get the applicable offset and scaling for the given raw value.
+
+        If data type only defines one set of offset/scaling then
+        the `raw_val` param can be omitted
+        """
+        if raw_val is None or len(self.segment_boundaries_raw) == 0:
+            try:
+                return self._offset[0], self._scale[0]  # type: ignore
+            except TypeError:
+                return self.offset, self.scale
+
+        return self._get_offset_scaling_from_list(raw_val, self.segment_boundaries_raw)
+
+    def get_offset_scaling_from_scaled(
+        self, scaled_val: Optional[float] = None
+    ) -> Tuple[float, float]:
+        """Get the applicable offset and scaling for the given scaled value.
+
+        If data type only defines one set of offset/scaling then
+        the `scaled_val` param can be omitted
+        """
+        if scaled_val is None or len(self.segment_boundaries_scaled) == 0:
+            try:
+                return self._offset[0], self._scale[0]  # type: ignore
+            except TypeError:
+                return self.offset, self.scale
+
+        return self._get_offset_scaling_from_list(
+            scaled_val, self.segment_boundaries_scaled
+        )
+
+    @property
+    def offset(self):
+        """Return first or only offset element"""
+        if isinstance(self._offset, list):
+            return self._offset[0]
+        else:
+            return self._offset
+
+    @offset.setter
+    def offset(self, offset: float) -> None:
```

Comment:
```suggestion
    def offset(self, offset: Union[float, List[float]]) -> None:
```

also, it would probably a good idea to rename the `offset` argument to e.g. `value` prevent shadowing of the function name...